Show post-install page on install or update event

MariagraziaAlastra opened the pull request:
<!-- Please add the WIP label if the PR isn't complete. -->

**Reviewer:** @laurengarcia 

<!-- Optional fields
**CC:**
**Depends on:** 
-->

## Description:
Show the new post-install page (same as in #47) on install.
The page is now live, which makes this easier to test :smile: 


## Steps to test this PR:
1. Build and reload extension
2. Page should be loaded in a new tab
3. Try reloading again from the interface - nothing should happen this time
4. Uninstall and install again - page should show
5. Close and reopen the browser - nothing should happen


## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation


###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

russellholt commented on shared/js/background.js:
```diff
@@ -69,9 +69,16 @@ function Background() {
 
   chrome.runtime.onInstalled.addListener(function(details) {
-    // only run the following section on install
+    // only run the following section on install and on update
```
doesn't this mean it will show on upgrade? we do not want to do that at this time.

laurengarcia replied:
I think you'll want to do 
`if (details.reason.match(/install/) { 
  settings.ready().then( () => { 
     settings.updateSetting( 'hasSeenPostInstall', true)
     chrome.tabs.update( ... )
}`

Double-check that everything above is correct but that's a rough idea of what should be happening here.

laurengarcia replied:
(Sorry for formatting above)

laurengarcia commented on shared/js/background.js:
```diff
@@ -69,9 +69,16 @@ function Background() {
+          chrome.tabs.create({
+            url: 'https://www.duckduckgo.com/app?post=1'
+          })
+        }
```
In addition to Russell's comment above, i also think `hasSeenPostInstall` should go into settings, and should also have a default value in shared/data/defaultSettings.js.

MariagraziaAlastra commented:
@russellholt Ok now it only runs on first install
@laurengarcia Addressed your comments as well, I agree having hasSeenPostInstall as a setting makes more sense 👍

jagtalon commented on shared/js/background.js:
```diff
@@ -69,11 +69,23 @@ function Background() {
 
+    // only show post install page on install
+    if (details.reason.match(/install/)) {
+        settings.ready().then( () => {
```
Wow arrow syntax you're living in the future! 🌵

laurengarcia commented:
Thanks!
